### PR TITLE
Feat(documentation): Use /// instead of //

### PR DESCRIPTION
**Pull request by RisingLeaf**
This PR addresses the change described in issue #10134

## Summary
All // in headers have been replaced with ///, single line // got another /// appended to be recognized.
draft because need to check if there are any errors in script (lots of lines changed)

## Screenshots
See -> files changed

## Usage examples
Use the existing doxygen script

## Testing Done
Does it build?
Tested doxygen script, comments get recognized

## Artwork Checklist
Code looks different but still ok

## Wiki Update
Still need update for style guide

## Performance Impact
Literally none

### CODE REVIEW

**Comment by RisingLeaf** [comment]
optionally I can place the empty /// in front of the comment

**Comment by TomGoodIdea** [comment]
> optionally I can place the empty /// in front of the comment

yeah, it would look better IMO

**Comment by RisingLeaf** [comment]
cant really argue about that, both looks a bit ugly to me, so here you go

File: source/AI.h [hunk 1]
```diff
@@ -42,62 +42,76 @@ class System;
...
 	AI(const PlayerInfo &player, const List<Ship> &ships,
 			const List<Minable> &minables, const List<Flotsam> &flotsam);
 
-	// Fleet commands from the player.
+	/// Fleet commands from the player.
+	///
 	void IssueFormationChange(PlayerInfo &player);
 	void IssueShipTarget(const std::shared_ptr<Ship> &target);
 	void IssueAsteroidTarget(const std::shared_ptr<Minable> &targetAsteroid);
 	void IssueMoveTarget(const Point &target, const System *moveToSystem);
```
Comment (warp-core) [inline source/AI.h]: This comment relates to all four of these function declarations, not just one of them.
Will Doxygen handle that or will it just associate it with the first method and then give the other nothing?

**Comment by RisingLeaf** [comment]
oops I also added the script, better remove it?

File: source/AI.h (hunk 1 above)
Comment (RisingLeaf) [reply source/AI.h]: with the first, it is kinda difficult to have a script recognize something like that

**Comment by RisingLeaf** [comment]
What I could also add is converting only top level comments in source files, meaning comments on functions

**Comment by RisingLeaf** [comment]
Script seems to work fine, I did not look at every change, be my guest if you want to do so, but I skimmed a good part of it and did not find any weirdness.

**Comment by tibetiroka** [comment]
For the style checker to accept the new format, the regex [here](https://github.com/endless-sky/endless-sky/blob/0c3d9cb8c76dac5f9aff3a46e12e9b4c8f5c393d/utils/check_code_style.py#L106) should be edited:
```
after_comment = re.compile("/?[^\\s#]")
```
I haven't tested this, but it should be fine.

**Comment by tibetiroka** [comment]
Try `"/?([^\\s#]|$)"`. Maybe that works with the empty comment lines.

**Comment by tibetiroka** [comment]
Oh wait, it's supposed to be a pattern that matches _bad_ input. I just forgot.
Did you try my first suggestion?
~~Also, you can just run it locally.~~

**Comment by RisingLeaf** [comment]
running local now, but still doesnt work

File: source/AI.h (hunk 1 above)
Comment (warp-core) [reply source/AI.h]: If Doxygen can't handle them, I don't think comments like this, introducing groups of methods, should be using the Doxygen syntax.

**Comment by tibetiroka** [comment]
`"/?[^/\\s#]"` works for me, though this won't give you an error for four slashes, I think.

File: source/AI.h (hunk 1 above)
Comment (RisingLeaf) [reply source/AI.h]: well using a script i have no real way of knowing if the comment describes a single function or all following

Comment (RisingLeaf) [reply source/AI.h]: if you have a good method tell me, but I would rather not go through all comments to look for this case

**Comment by mOctave** [comment]
While I'd like to see Doxygen support, ~~I'm not sure programatically converting comments is the way to go here, since it would still get only a small fraction of the codebase documented and will almost certainly end up getting stuck on edge cases no matter which way it gets done. Which means that someone would still have to skim through all 5000 lines that have been changed and manually evaluate them to make sure they make sense, at which point why don't we just manually add the comments in the first place?~~

Yes, I do realize that I'm suggesting that somebody add or modify several thousand comments in 100+ files. I think this could become some sort of community project, actually. I'm sure there's plenty of people (myself included) here or on Discord with some minimal level of coding experience who would be happy to pitch in and add comments for a file or two.

After actually reading the diff and noticing that this is only being done to header files, some of my concerns are reduced. But still, why are we only doing this to header files, which we're then going to have to massively bloat if we want to fully document everything? I stand by my earlier suggestion for a community project with brief descriptions in headers and full documentation in code, although I think this approach could work for a first pass.

File: source/AI.h (hunk 1 above)
Comment (tibetiroka) [reply source/AI.h]: I don't know of a better option, but I agree that this shouldn't be in doxygen.

Comment (RisingLeaf) [reply source/AI.h]: so apparently you can do groups in doxygen with:
///@{
void funca();
void funcb();
///@}

but we still have to go through each file

Comment (RisingLeaf) [reply source/AI.h]: Proposal when we know what we want to do about this, we make a checklist of files and whenever someone feels like it, they can go through a few, suggest changes and then we mark them of the list.
Or my preferred option: Just don't care about it, but that is up to you.

**Comment by RisingLeaf** [comment]
So after what moctave wrote here and in private to me, following suggestion for going forward:
1.
 - a) This PR gets merged, even if some things are not quite right yet (comments for groups of functions for example)
 - b) At the same time changes to the style guide are made explaining how doxygen comments are going to be used
 - 1a and 1b should happen simultaneously
2. An issue gets created with a checklist of some sort showing which files have already been looked at and which ones need work
3. People create PRs adjusting comments for individual files or multiple, marking them of the checklist

This has some benefits over just doing everything in one PR, those being
 - This PR does not need a whole lot of review but moves a lot of line changes out of the way and makes latter PRs easier to look through
 - the issue would be a perfect place to point first time contributors towards, while also being open for anybody else: you are currently editing AI.h anyways? why not quickly adjust the comments there
 - Doxygen comments do not need to be perfect right away, but some benefit will already provided through this PR

**Comment by warp-core** [comment]
> you are currently editing AI.h anyways? why not quickly adjust the comments there

I suggest that users who are looking to add or edit a small number of functions should not be encouraged to update every single comment in the entire file.
In general, it is undesirable to modify comments in PRs that are doing something completely unrelated, except for happening to be in the same file.

I will not approve this PR in its current state. Comments applying to groups of declarations should either be handled correctly in Doxygen, or excluded from Doxygen syntax altogether. I strongly prefer the former over trying to do it piecemeal. I believe not being able to see the end product as a whole will be detrimental to trying to figure out which style we want to pursue, and having the comment style in some files be different to others will leave other contributors conflicted: "I'm adding new functions to this file. It hasn't been updated with the new comment style. Should my new comments use that style or be consistent with the existing content of this file, since someone has to update the rest of it anyway." Or, even "I'll just do all the comments even though I'm only otherwise editing one line here."

You may distribute the work of addressing this as you wish. I don't care if one person does 100% of the work, or 100 people each do 1% (though, that'd be a massive commit description). Though, I feel the amount of work it would take is being severely overexaggerated.

Additionally, it would be extremely helpful in reviewing this to actually see the documentation that Doxygen produces from it.

**Comment by RisingLeaf** [comment]
> > you are currently editing AI.h anyways? why not quickly adjust the comments there
> 
> I suggest that users who are looking to add or edit a small number of functions should not be encouraged to update every single comment in the entire file. In general, it is undesirable to modify comments in PRs that are doing something completely unrelated, except for happening to be in the same file.

that was not what I suggested here, what I thought of is you quickly open a new branch, do the comments and return to your old branch

> I will not approve this PR in its current state. Comments applying to groups of declarations should either be handled correctly in Doxygen, or excluded from Doxygen syntax altogether. I strongly prefer the former over trying to do it piecemeal. I believe not being able to see the end product as a whole will be detrimental to trying to figure out which style we want to pursue,

I definitely wont finish this before there is a decision on what style is going to be used.

> You may distribute the work of addressing this as you wish. I don't care if one person does 100% of the work, or 100 people each do 1% (though, that'd be a massive commit description). Though, I feel the amount of work it would take is being severely overexaggerated.

If you feel like that, you are welcome to go through all files and suggest changes. I wont because I do not have the time to do so.

> Additionally, it would be extremely helpful in reviewing this to actually see the documentation that Doxygen produces from it.

Grab the branch, and run doxygen, I wont upload the whole doxygen generated docs everytime I change something.

**Pull request closed without merging**